### PR TITLE
Detect signals

### LINKED ISSUE #55: Detect sigint and reap output device
If you restart the service, it will continue to create new virtual output devices.

### PR DESCRIPTION
**Pull request by donniebreve**
Attempts to detect interrupt signals and handle them.
 
#56 should be merged first.

Fixes #55

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -10,25 +12,80 @@
...
+{
+    if (signal == SIGHUP)
+    {
+        should_reload = 1;
+    }
+    else if (signal == SIGINT || signal == SIGTERM)
+    {
+        should_exit = 1;
+    }
+    else if (signal == SIGSEGV)
```
Comment (auouymous) [inline src/main.c:31]: Why catch SIGSEGV, do nothing and not properly exit with EXIT_FAILURE?

File: src/binding.c [hunk 2]
```diff
@@ -56,10 +56,20 @@ int bindInput(char* eventPath)
     return EXIT_SUCCESS;
 }
 
+/**
+ * Releases the input device.
+ * */
+int release_input(char* eventPath)
```
Comment (auouymous) [inline src/binding.c:62]: Should this write a message to stdout with the unused eventPath parameter like is done in bind_input()?

File: src/binding.c [hunk 3]
```diff
@@ -107,3 +117,13 @@ int bindOutput()
     fprintf(stdout, "info: successfully created virtual output device\n");
     return EXIT_SUCCESS;
 }
+
+/**
+ * Releases the virtual output device.
+ * */
+int release_output()
```
Comment (auouymous) [inline src/binding.c:124]: Should this write a message to stdout like bind_output()?

File: src/main.c [hunk 4]
```diff
@@ -39,18 +96,37 @@ int main(int argc, char* argv[])
...
         }
         if (result == (ssize_t)0)
         {
-            return ENOENT;
+            return EXIT_FAILURE;
         }
         if (result != sizeof(inputEvent))
         {
-            return EIO;
+            return EXIT_FAILURE;
```
Comment (auouymous) [inline src/main.c:129]: Should these exits try to release the devices?

**Review by auouymous — CHANGES_REQUESTED** [review]
This PR contains the commit from #56.

File: src/main.c (hunk 1 above)
Comment (donniebreve) [reply src/main.c:31]: I don't believe there's anything useful I can do and hiding the segmentation fault with EXIT_FAILURE isn't ideal. I'm removing the handler.

File: src/binding.c (hunk 2 above)
Comment (donniebreve) [reply src/binding.c:62]: Added

File: src/binding.c (hunk 3 above)
Comment (donniebreve) [reply src/binding.c:124]: I would like to figure out how to find the virtual device information (like event path) first.

Comment (donniebreve) [reply src/binding.c]: Added

File: src/main.c (hunk 4 above)
Comment (donniebreve) [reply src/main.c]: They should, and now do.

**Comment by auouymous** [comment]
Why do the second two patches delete massive chunks only to insert them mostly unchanged?

**Comment by auouymous** [comment]
Ah, thank you. I can't resolve my review comments but everything looks good.

**Pull request merged**